paymentsheet/src/main/java/com/stripe/android/common/ui/BottomSheet.kt

```diff
@@ -0,0 +1,128 @@
+import kotlinx.coroutines.flow.first
+
+@OptIn(ExperimentalMaterialApi::class)
+internal class BottomSheetState(
+    val modalBottomSheetState: ModalBottomSheetState,
+) {
+
+    suspend fun show() {
+        modalBottomSheetState.show()
+    }
+
+    suspend fun awaitDismissal() {
+        snapshotFlow { modalBottomSheetState.isVisible }.first { isVisible -> !isVisible }
+    }
+
+    suspend fun hide() {
+        modalBottomSheetState.hide()
+    }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun rememberBottomSheetState(
+    confirmValueChange: (ModalBottomSheetValue) -> Boolean = { true },
+): BottomSheetState {
+    val modalBottomSheetState = rememberModalBottomSheetState(
+        initialValue = ModalBottomSheetValue.Hidden,
+        confirmValueChange = confirmValueChange,
+        skipHalfExpanded = true,
+        animationSpec = tween(),
+    )
+
+    return remember {
+        BottomSheetState(modalBottomSheetState)
+    }
+}
+
+/**
+ * Renders the provided [sheetContent] in a modal bottom sheet.
+ *
+ * @param state The [BottomSheetState] that controls the visibility of the bottom sheet.
+ * @param onShow Called when the bottom sheet is displayed for the first time. This might be used to
+ * navigate to a specific screen.
+ * @param onDismissed Called when the user dismisses the bottom sheet by swiping down. You should
+ * inform your view model about this change.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun BottomSheet(
+    state: BottomSheetState,
+    modifier: Modifier = Modifier,
+    onDismissed: () -> Unit,
+    onShow: () -> Unit = {},
+    sheetContent: @Composable ColumnScope.() -> Unit,
+) {
+    val systemUiController = rememberSystemUiController()
+    val scrimColor = ModalBottomSheetDefaults.scrimColor
+
+    val isExpanded = state.modalBottomSheetState.targetValue == Expanded
+
+    val statusBarColorAlpha by animateFloatAsState(
+        targetValue = if (isExpanded) scrimColor.alpha else 0f,
+        animationSpec = tween(),
+        label = "StatusBarColorAlpha",
+    )
+
+    LaunchedEffect(Unit) {
+        state.show()
+        onShow()
+
+        state.awaitDismissal()
+        onDismissed()
+    }
+
+    DisposableEffect(systemUiController, statusBarColorAlpha) {
+        systemUiController.setStatusBarColor(
+            color = scrimColor.copy(statusBarColorAlpha),
+            darkIcons = false,
+        )
+
+        onDispose {}
+    }
+
+    DisposableEffect(systemUiController) {
+        systemUiController.setNavigationBarColor(
+            color = Color.Transparent,
+            darkIcons = false,
+        )
+
+        onDispose {}
+    }
+
+    ModalBottomSheetLayout(
+        modifier = modifier.statusBarsPadding(),
+        sheetState = state.modalBottomSheetState,
+        sheetContent = {
+            sheetContent()
+            Spacer(modifier = Modifier.windowInsetsBottomHeight(WindowInsets.systemBars))
```

Comment:
Have to apply this in `sheetContent` instead of on the `Modifier` so that content can scroll below the navigation bar.

Reply:
Would be cool if we had a way to screenshot test stuff like this.